includes/blocks/class-sensei-block-contact-teacher.php

```diff
@@ -46,20 +46,72 @@ public function register_block() {
 	 */
 	public function render_contact_teacher_block( $attributes, $content ): string {
 		global $post;
-		$href = add_query_arg( array( 'contact' => $post->post_type ) );
 
-		return wp_kses_post( $this->add_href_attribute( $content, $href ) );
+		if ( ! empty( Sensei()->settings->settings['messages_disable'] ) || ! is_user_logged_in() ) {
+			return '';
+		}
+
+		$contact_form_link = add_query_arg( array( 'contact' => $post->post_type ) );
+
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Arguments used for comparison.
+		$message_sent = ( isset( $_GET['send'] ) && 'complete' === $_GET['send'] );
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Arguments used for comparison.
+		$contact_form_open = isset( $_GET['contact'] ) || $message_sent;
+
+		$contact_form = $this->teacher_contact_form( $post, $message_sent );
+
+		return '<div id="private_message" class="sensei-collapsible">
+				' . ( $this->add_button_attributes( $content, $contact_form_link ) ) . '
+				<div class="sensei-collapsible__content ' . ( $contact_form_open ? '' : 'collapsed' ) . '">' . $contact_form . '</div>
+			</div>';
+	}
+
+	/**
+	 * Render contact teacher form.
+	 *
+	 * @param WP_Post $post         The current post.
+	 * @param bool    $message_sent Display message sent feedback.
+	 *
+	 * @return string
+	 */
+	private function teacher_contact_form( $post, $message_sent ) {
+
+		$confirmation = '';
+		if ( $message_sent ) {
+
+			$confirmation_message = __( 'Your private message has been sent.', 'sensei-lms' );
```

Comment:
WDYT to move this message to outside the collapsible and makes it collapsed by default when the message was sent? The current behavior is not bad, but I think it could be a little more natural in this way.

Reply:
Moved things around @ 8d5b759